src/System.IO.Packaging/tests/Tests.cs

```diff
@@ -3639,6 +3639,109 @@ public void SetEmptyPropertyToNull()
             }
         }
 
+        [Fact]
+        public void CreateWithFileAccessWrite()
+        {
+            string[] fileNames = new [] { "file1.txt", "file2.txt", "file3.txt" };
+
+            using (var stream = new MemoryStream())
+            {
+                using (var package = Package.Open(stream, FileMode.Create, FileAccess.Write))
+                {
+                    foreach( var fileName in fileNames)
+                    {
+                        PackagePart part = package.CreatePart(PackUriHelper.CreatePartUri(new Uri(fileName, UriKind.Relative)),
+                                                              System.Net.Mime.MediaTypeNames.Text.Plain,
+                                                              CompressionOption.Fast);
+                        using (var writer = new StreamWriter(part.GetStream(), Encoding.ASCII))
+                        {
+                            // just write the filename as content
+                            writer.Write(fileName);
+                        }
+                    }
+                }
+
+                // reopen for read and validate the content
+                stream.Seek(0, SeekOrigin.Begin);
+                using (var readPackage = Package.Open(stream))
+                {
+                    foreach(var fileName in fileNames)
+                    {
+                        var part = readPackage.GetPart(PackUriHelper.CreatePartUri(new Uri(fileName, UriKind.Relative)));
+
+                        using (var partStream = part.GetStream())
+                        using (var reader = new StreamReader(partStream, Encoding.ASCII))
+                        {
+                            Assert.Equal(fileName.Length, partStream.Length);
+                            Assert.Equal(fileName, reader.ReadToEnd());
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        public void VeryLargePart()
+        {
+            // FileAccess.Write is important, this tells ZipPackage to open the underlying ZipArchive in 
+            // ZipArchiveMode.Create mode as opposed to ZipArchiveMode.Update
+            // When ZipArchive is opened in Create it will write entries directly to the zip stream
+            // When ZipArchive is opened in Update it will write uncompressed data to memory until 
+            // the archive is closed.
+            using (var stream = new MemoryStream())
```

Comment:
Should we use a FileStream rather than having a 6GB stream in memory?

Reply:
It'll never be 6GB of memory, that's the whole point.  Only thing in memory here will be compressed data.  I measured and its under 70MB.  Still I made it outerloop because it takes a few seconds.

Reply:
Ah, ok, thanks, I misread.